src/util/string.h

```diff
@@ -473,6 +473,23 @@ inline void str_replace(std::string &str, const std::string &pattern,
 	}
 }
 
+ /**
++ * Replace all occurrences of \p pattern in \p str with \p replacement.
++ *
++ * @param str String to replace pattern with replacement within.
++ * @param pattern The pattern to replace.
++ * @param replacement What to replace the pattern with.
++ */
+inline void str_replace(std::wstring &str, const std::wstring &pattern,
```

Comment:
You could make this function return the amount of replaced parts (amount of loops, so to say).
This way it your string duplication and `compare()` call would become superfluous.

Just as an idea, because I think that's a cleaner solution.

Reply:
Just take a look at line 466